saga/seata-saga-statemachine-designer/src/spec/style/Node.js

```diff
@@ -24,9 +24,381 @@ export default class Node extends BaseSpec {
+        }];
+      } else {
+        waypoints1 = [{
+          x: sourceX + 18,
+          y: sourceY,
+        }, {
+          x: targetX + 18,
+          y: (targetY + 36) + 20,
+        }, {
+          x: targetX + 18,
+          y: targetY + 36,
+        }];
+      }
+      startState.catch.edge = assign(startState.catch.edge || {}, {
+        [option.Next]: {
+          style: {
+            waypoints: waypoints1,
+            source: startState.Name,
+            target: option.Next,
+          },
+          Type: 'ExceptionMatch',
+        },
+      });
+    }
+    this.importJson(startState.catch);
+  }
+
+  addCatch(definitions, node, catchList, adjList) {
+    node.catch = {};
+    if (node.Catch) {
+      const {
+        style: {
+          bounds: {
+            x,
+            y,
+          },
+        },
+      } = node;
+      const newX = x;
+      const newY = y;
+      node.catch.style = {};
+      node.catch.style.bounds = {
+        x: newX + 50,
+        y: newY - 20,
+        width: 36,
+        height: 36,
+      };
+    }
+    this.importJson(node.catch);
+
+    let prev = node.catch;
+    let width1;
+    let height1;
+    catchList.get(node).forEach((semantic) => {
+      if ((semantic.Type === 'ServiceTask') || (semantic.Type === 'ScriptTask') || (semantic.Type === 'SubStateMachine')) {
+        width1 = 100;
+        height1 = 80;
+      } else {
+        width1 = 36;
+        height1 = 36;
+      }
+      semantic.style = {};
+      semantic.style.bounds = {
+        x: prev.style.bounds.x - 50,
+        y: prev.style.bounds.y - 100,
+        width: width1,
+        height: height1,
+      };
+      prev = semantic;
+
+      this.importStates(definitions, semantic, null, adjList);
+    });
+  }
+
+  importStates(definitions, startState, begin, adjList) {
+    const visited = [];
+    const queue = [];
+    if (begin !== null) {
+      if (startState.style === undefined) {
+        startState.style = {};
+        if (startState.style.bounds === undefined) {
+          startState.style.bounds = {
+            x: begin.style.bounds.x + 150, // Adjust x-coordinate
+            y: begin.style.bounds.y, // Adjust y-coordinate
+            width: 100,
+            height: 80,
+          };
+        }
+      }
+      this.importJson(startState);
+    }
+    queue.push(startState);
+
+    function setBounds(neighbor, x, y, width1, height1) {
+      if (neighbor.style.bounds === undefined) {
+        neighbor.style.bounds = {
+          x, // Adjust x-coordinate
+          y, // Adjust y-coordinate
+          width: width1,
+          height: height1,
+        };
+        visited.push([x, y]);
+      }
+    }
+
+    while (queue.length) {
+      const currentState = queue.shift();
+
+      adjList.get(currentState).forEach((neighbor) => {
+        if (neighbor.style === undefined) {
+          neighbor.style = {};
+          if ((neighbor.Type === 'Fail') || (neighbor.Type === 'Succeed')) {
```

Comment:
Use `is`

Reply:
Here we cant use is function as it has only three conditions related to task